internal/services/automation/automation_variables_data_source.go

```diff
@@ -0,0 +1,289 @@
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"resource_group_name": commonschema.ResourceGroupName(),
+
+			"automation_account_name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: validate.RunbookName(),
+			},
+
+			"bool": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					// How do I make this related to the specific type?
+					Schema: map[string]*pluginsdk.Schema{
+						"name": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
+						"description": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"encrypted": {
+							Type:     pluginsdk.TypeBool,
+							Computed: true,
+						},
+
+						"value": {
+							Type:     pluginsdk.TypeBool,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"datetime": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					// How do I make this related to the specific type?
+					Schema: map[string]*pluginsdk.Schema{
+						"name": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
+						"description": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"encrypted": {
+							Type:     pluginsdk.TypeBool,
+							Computed: true,
+						},
+
+						"value": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"encrypted": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					// How do I make this related to the specific type?
+					Schema: map[string]*pluginsdk.Schema{
+						"name": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
+						"description": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"encrypted": {
+							Type:     pluginsdk.TypeBool,
+							Computed: true,
+						},
+
+						"value": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"int": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					// How do I make this related to the specific type?
+					Schema: map[string]*pluginsdk.Schema{
+						"name": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
+						"description": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"encrypted": {
+							Type:     pluginsdk.TypeBool,
+							Computed: true,
+						},
+
+						"value": {
+							Type:     pluginsdk.TypeInt,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"null": {
```

Comment:
What variable type is this? I don't see a resource for this variable type - so in what cases would this be populated?

Reply:
Microsoft offer a `null` [type variable](https://learn.microsoft.com/en-us/azure/automation/shared-resources/variables?tabs=azure-powershell#variable-types) in an automation account. Currently there is no TF resource for it, but it can be populated from the Azure portal. I added to this data source to try and get all the different variables that might be created available.

If we want consistency, I could add a resource for the null variable type so we can create them in TF.